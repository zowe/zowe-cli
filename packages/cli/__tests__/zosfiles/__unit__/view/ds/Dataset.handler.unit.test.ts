/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

import { Get } from "@zowe/zos-files-for-zowe-sdk";
import { UNIT_TEST_ZOSMF_PROF_OPTS } from "../../../../../../../__tests__/__src__/TestConstants";

describe("View data set handler", () => {
    describe("process method", () => {
        it("should view a single data set", async () => {
            // Require the handler and create a new instance
            const handlerReq = require("../../../../../src/zosfiles/view/ds/Dataset.handler");
            const handler = new handlerReq.default();
            const dataSetName = "testing";


            // Vars populated by the mocked function
            let apiMessage = "";
            let jsonObj;
            let logMessage = "";
            let fakeSession = null;

            // Mock the view ds function
            Get.dataSet = jest.fn(async (session) => {
                fakeSession = session;
                return Buffer.from("downloaded");
            });

            // Mocked function references
            const profFunc = jest.fn((_args) => {
                return {
                    host: "fake",
                    port: "fake",
                    user: "fake",
                    password: "fake",
                    auth: "fake",
                    rejectUnauthorized: "fake",
                };
            });

            try {
                // Invoke the handler with a full set of mocked arguments and response functions
                await handler.process({
                    arguments: {
                        $0: "fake",
                        _: ["fake"],
                        dataSetName,
                        ...UNIT_TEST_ZOSMF_PROF_OPTS
                    },
                    response: {
                        data: {
                            setMessage: jest.fn((setMsgArgs) => {
                                apiMessage = setMsgArgs;
                            }),
                            setObj: jest.fn((setObjArgs) => {
                                jsonObj = setObjArgs;
                            })
                        },
                        console: {
                            log: jest.fn((logArgs) => {
                                logMessage += "\n" + logArgs;
                            })
                        },
                        progress: {
                            startBar: jest.fn((_parms) => {
                                // do nothing
                            }),
                            endBar: jest.fn(() => {
                                // do nothing
                            })
                        }
                    },
                    profiles: {
                        get: profFunc
                    }
                } as any);
            } catch (e) {
                // Do nothing
            }

            expect(Get.dataSet).toHaveBeenCalledTimes(1);
            expect(Get.dataSet).toHaveBeenCalledWith(fakeSession, dataSetName, {
                task: {
                    percentComplete: 0,
                    stageName: 0,
                    statusMessage: "Retrieving data set"
                }
            });
            expect(jsonObj).toMatchSnapshot();
            expect(apiMessage).toMatchSnapshot();
            expect(logMessage).toMatchSnapshot();
        });
    });
});
