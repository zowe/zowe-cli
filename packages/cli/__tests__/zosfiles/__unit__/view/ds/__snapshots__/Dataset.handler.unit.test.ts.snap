// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View data set handler process method should view a single data set 1`] = `
{
  "apiResponse": {},
  "commandResponse": "downloaded",
  "success": true,
}
`;

exports[`View data set handler process method should view a single data set 2`] = `""`;

exports[`View data set handler process method should view a single data set 3`] = `
"
downloaded"
`;
